sdk/core/core-amqp/test/retry.spec.ts

```diff
@@ -203,6 +203,77 @@ dotenv.config();
       }
     });
 
+    it("should not sleep after final failure if all attempts return a retryable error (no retries)", async function() {
+      let counter = 0;
+      // Create an abort controller so we can clean up the delay's setTimeout ASAP after the race.
+      const delayAbortController = new AbortController();
+      try {
+        const config: RetryConfig<any> = {
+          operation: async () => {
+            debug("counter: %d", ++counter);
```

Comment:
Needed?

Reply:
Incrementing the counter is needed, but the debug call is not (just copy pasted from another test)

Reply:
And...updated